### DISCUSSION TITLE
It is possible to use OAuth + Cookie and Username/Password + Bearer

### DISCUSSION THREAD

**Discussion by KentonParton** [fastapi-users/fastapi-users#960] [Q&A] [answered]
I have been using FastAPI Users OAuth/Cookie/JWT for a few months and it has been working great! 

Now, I would like to add Username_Password/ Bearer/ JWT. The reason for adding Username_Password/ Bearer/ JWT is to make it easier for server based interactions with my API.

How would I use both side-by-side? The "current_active_user" seems to be the holdup. Probably have to add some logic to check whether Bearer exists or cookie exists?

Appreciate the help!

**Comment by frankie567** [comment, marked as answer]
Actually, this should work out-of-the-box. The `current_user` dependency is able to handle several authentication backends and try them one by one until one matches (or not, and if this case, the user is not authenticated). So let's say you have two backends:

```py
cookie_backend = AuthenticationBackend(
    name="cookie",
    transport=cookie_transport,
    get_strategy=get_jwt_strategy,
)

bearer_backend = AuthenticationBackend(
    name="bearer",
    transport=bearer_transport,
    get_strategy=get_jwt_strategy,
)
```

You just have to declare them both on `FastAPIUsers` class:

```py
fastapi_users = FastAPIUsers(
    get_user_manager,
    [cookie_backend, bearer_backend],
    User,
    UserCreate,
    UserUpdate,
    UserDB,
)
```

Don't forget also to add [authentication routes for both of them](https://fastapi-users.github.io/fastapi-users/configuration/routers/auth/), so the user can either get a cookie or a bearer token.

**Reply by KentonParton** [reply]
Agg geez, I should have spotted that. Thanks @frankie567 😃
